lib/developer_portal/app/views/developer_portal/applications/show.html.liquid

```diff
@@ -140,16 +140,23 @@
                       {% endfor %}
                     </table>
                     <p class='sub'>These are application keys used to authenticate requests.</p>
+
+                    {% if application.can.add_application_keys? %}
+                      {% assign enabled_block_style =  "" %}
+                      {% assign disabled_block_style =  "display: none;" %}
+                    {% else %}
+                      {% assign enabled_block_style =  "display: none;" %}
+                      {% assign disabled_block_style =  "" %}
+                    {% endif %}
+                    
                     <div class="operations">
-                      {% if application.can.add_application_keys? %}
```

Comment:
So this block was causing the failing behavior. The problem is that either the `enabled_block` or `disabled_block` was not even rendered, so the switch didn't work, because what it does is shows/hides the blocks.

The change in this PR fixes it. However, this is the default liquid block, so the fix will not propagate to the existing dev portals.
A workaround for that would be to change the `create.js.erb` and `destroy.js.erb` to add the missing block elements. But I think it's not a big deal, because even if we have customers complaining about this, we can provide the snippets that will fix the behavior.